src/Nethermind/Nethermind.Core/Extensions/Bytes.cs

```diff
@@ -602,26 +602,54 @@ public static string ByteArrayToHexViaLookup32Safe(byte[] bytes, bool withZeroX)
             
             int length = bytes.Length * 2 + (withZeroX ? 2 : 0);
             StateSmall stateToPass = new StateSmall(bytes, withZeroX);
+            
             return string.Create(length, stateToPass, (chars, state) =>
             {
-                int offset0x = 0;
+                ref var charsRef = ref MemoryMarshal.GetReference(chars);
+
                 if (state.WithZeroX)
                 {
-                    chars[0] = '0';
-                    chars[1] = 'x';
-                    offset0x += 2;
+                    charsRef = '0';
+                    Unsafe.Add(ref charsRef, 1) = 'x';
+                    charsRef = Unsafe.Add(ref charsRef, 2);
+                }
+
+                ref var input = ref state.Bytes[0];
+                ref var output = ref Unsafe.As<char, uint>(ref charsRef);
+
+                int toProcess = state.Bytes.Length;
+
+                const int unroll = 8;
+
+                while (toProcess > unroll)
+                {
+                    output = Lookup32[input];
+                    Unsafe.Add(ref output, 1) = Lookup32[Unsafe.Add(ref input, 1)];
```

Comment:
Unsafe.Add(ref output, 2 + prefixLength0x) = Lookup32[Unsafe.Add(ref input, 2)];

Reply:
output is amended in lines above